.github/workflows/test-spm.yml

```diff
@@ -0,0 +1,72 @@
+name: Test SPM
+
+on:
+  push:
+    branches: stable
+    paths:
+      - '.github/workflows/test-spm.yml'
+      - '**/*.swift'
+  pull_request:
+    paths:
+      - '.github/workflows/test-spm.yml'
+      - '**/*.swift'
+
+jobs:
+  linux:
+    name: Linux
+    runs-on: ubuntu-latest
+    container: swiftgen/swift:5.3
+    steps:
+      - 
+        name: Checkout
+        uses: actions/checkout@v2
+      - 
+        # Note: we can't use `ruby/setup-ruby` on custom docker images, so we
+        # have to do our own caching
```

Comment:
👍

Reply:
Maybe worth exploring in the future to how to create our own reusable GitHub actions but compatible with custom docker images? But no need to do that now, just a thought to keep in mind for one day

Reply:
Here's one issue I found BTW, which gives some extra info. GitHub does a bunch of custom stuff in their ubuntu images which we don't have access to. If at some point they release some sort of Dockerfile, we can maybe base our images on theirs.